torch/testing/_internal/common_fsdp.py

```diff
@@ -1352,6 +1355,41 @@ def _test_fsdp_parity(
             )
 
 
+def test_compiled_fsdp():
+    class FullyShardPatch(Enum):
+        @staticmethod
+        def fully_shard_with_compiled_compute(*args, **kwargs):
+            args[0].forward = torch.compile(args[0].forward)
+            return torch.distributed._composable.fsdp.fully_shard(*args, **kwargs)
+
+        @staticmethod
+        def fully_shard_with_full_compile(*args, **kwargs):
```

Comment:
I am not sure if this makes sense since for full compile, we would only apply it to the root module. However, IIUC, this would do like `torch.compile(fully_shard(module))` for each `fully_shard` call, including non-root.

Reply:
good catch! will think about a better way but will remove ``fully_shard_with_full_compile`` from this PR